packages/scene-composer/src/components/panels/scene-settings/SceneTagSettingsEditor.tsx

```diff
@@ -37,62 +36,38 @@ export const SceneTagSettingsEditor: React.FC = () => {
           ...getSceneProperty(KnownSceneProperty.ComponentSettings),
           [KnownComponentType.Tag]: newTagSettings,
         };
-
         setSceneProperty(KnownSceneProperty.ComponentSettings, newComponentSettings);
       }
       setTagSettings(newTagSettings);
     },
-    [tagSettings, getSceneProperty, setSceneProperty],
+    [internalScale, tagSettings, getSceneProperty, setSceneProperty],
   );
 
   const onSliderChange = useCallback(
     (event) => {
       setInternalScale(Number(event.target.value));
-      setDirty(true);
     },
-    [setInternalScale, setDirty],
+    [internalScale, tagSettings],
   );
 
-  const onInputBlur = useCallback(() => {
-    // If slider is getting focus, this makes sure to execute setFocusSlider(true) first to keep showing it.
-    const id = setTimeout(() => {
-      setFocusInput(false);
-    }, 1);
-    return () => {
-      clearTimeout(id);
-    };
-  }, [setFocusInput]);
-
   const onInputChange = useCallback(
     (event) => {
       let value = event.detail.value;
       if (value < 0) {
         value = 0;
       }
-
       if (value !== internalScale) {
         setInternalScale(Number(value));
-        setDirty(true);
       }
     },
-    [setInternalScale, setDirty],
+    [internalScale, tagSettings],
```

Comment:
Why do you add `tagSettings` to all of these useCallback functions as a dependency?